lib/dice/src/handoff.rs

```diff
@@ -13,6 +16,25 @@ use serde::{Deserialize, Serialize};
 // TODO: get from app.toml -> chip.toml at build time
 const MEM_START: usize = 0x4010_0000;
 const ALIAS_START: usize = MEM_START;
+const ALIAS_SIZE: usize = 0x800;
+const SPMEASURE_START: usize = ALIAS_START + ALIAS_SIZE;
+const SPMEASURE_SIZE: usize = 0x800;
+const RNG_START: usize = SPMEASURE_START + SPMEASURE_SIZE;
+const RNG_SIZE: usize = 0x100;
+
+pub fn slice_from_parts(start: usize, size: usize) -> &'static mut [u8] {
```

Comment:
This seems to only be used in 2 places.  I'd much rather see the safety comment in those two places.

Reply:
This function should also be marked `unsafe` itself. Currently it claims to present a safe interface, which is not correct since it has no way to check that `start`/`size` are reasonable.

Reply:
This is an interesting place where my efforts to consolidate implementation into traits / functions complicates things. I went back to the rustnomicon this morning to remind myself of the details and AFAIK the whole `HandoffData` trait here is unsafe since it can't determine whether types implementing the trait have overlapping memory regions. This is going to take some refactoring.

Reply:
In refactoring the handoff module I've:
* converted all const START / SIZE values to be const Range<usize>
* added static assertions to prevent overlap of these regions as well as ensuring their size is sufficient
* clarified safety comments to better align with conventions from the standard library writers guide & the rustonomicon